### PR TITLE
Ensure CRDs are created on boot

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Make sure CRDs exist early in bootup, as we require these CRDs to be present in order to launch controllers.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -345,6 +345,15 @@ func main() {
 	}
 	setupLog.WithValues("provider", provider).Info("Checking type of cluster")
 
+	// If configured to managed CRDs, do a preliminary install of them here. The Installation controller
```
Comment (coutinhop) [inline cmd/main.go:348]:
```suggestion
	// If configured to manage CRDs, do a preliminary install of them here. The Installation controller
```

File: cmd/main.go [hunk 2]
```diff
@@ -269,6 +274,20 @@ func main() {
 		os.Exit(1)
 	}
 
+	// If configured to managed CRDs, do a preliminary install of them here. The Installation controller
```
Comment (coutinhop) [inline cmd/main.go:277]: typo: managed -> manage?

**Review by coutinhop — APPROVED** [review]

File: cmd/main.go [hunk 3]
```diff
@@ -269,6 +274,20 @@ func main() {
...
+	// will reconcile them as well, but we need to make sure they are installed before we start the rest of the controllers.
+	if bootstrapCRDs || manageCRDs {
```
Comment (rene-dekker) [inline cmd/main.go:279]: If I understand it correctly both  bootstrapCRDs and manageCRDs will install the OSS CRDs on startup. When manageCRDs is specified, the core controller will also apply the CRDs based on the variant.
What I don't fully understand at this point is that if my memory serves me right, the controller runtime requires a full restart in order to use new CRDs. In the case of enterprise, how can we use these flags to allow us to stop applying manifests manually?

Comment (caseydavenport) [reply cmd/main.go:279]: > What I don't fully understand at this point is that if my memory serves me right, the controller runtime requires a full restart in order to use new CRDs.

I'll try this out.. I didn't notice this problem when testing my changes with OSS Calico.

Comment (caseydavenport) [reply cmd/main.go:279]: I sort of assume we would have handled that already with the existing `manageCRDs` logic though.

Comment (rene-dekker) [reply cmd/main.go:279]: I checked and didn't see a restart in the core controller.

Comment (rene-dekker) [reply cmd/main.go:279]: There is this logic later in the reconcile loop that may solve the issue I brought up: https://github.com/tigera/operator/blob/master/pkg/controller/installation/core_controller.go#L968

**Review by rene-dekker — APPROVED** [review]
lgtm

**Pull request merged**